clang/lib/StaticAnalyzer/Core/ExprEngine.cpp

```diff
@@ -2801,8 +2799,35 @@ void ExprEngine::processBranch(const Stmt *Condition,
     if (StTrue && StFalse)
       assert(!isa<ObjCForCollectionStmt>(Condition));
 
-    if (StTrue)
-      Builder.generateNode(StTrue, true, PredN);
+    if (StTrue) {
+      // If we are processing a loop condition where two iterations have
+      // already been completed and the the false branch is also feasible, then
+      // don't assume a third iteration, because it is a redundant execution
```

Comment:
```suggestion
      // don't assume a third iteration because it is a redundant execution
```
IDK, but Grammarly told me so :D